rswag-specs/lib/rswag/specs/swagger_formatter.rb

```diff
@@ -99,17 +106,17 @@ def yaml_prepare(doc)
         JSON.parse(json_doc)
       end
 
-      def metadata_to_swagger(metadata)
-        response_code = metadata[:response][:code]
-        response = metadata[:response].reject { |k, _v| k == :code }
+      def swagger_path(response, operation, path_item)
```

Comment:
This looks a little bit confusing to me as on this method you've changed it to have params where the order is `response` then `operation`, but in `upgrade_response_produces!` below you have `operation`, then `response`.

See comment above - not sure you need to "eagerly" extract out the things to local vars, might be simpler to keep passing the metadata and extract the items you need lazily? Either that or introduce a simple data structure internally that can be passed around, rather than the three separate things?